lib/utils/property-references.js

```diff
@@ -96,6 +96,12 @@ function definePropertyReferenceExtractor(context) {
   /** @type {{ toRefNodes: Set<ESNode>, toRefsNodes: Set<ESNode>} | null} */
   let toRefSet = null
 
+  let isFunctionalTemplate = false
+  const templateBody = context.getSourceCode().ast.templateBody
+  if (templateBody) {
+    isFunctionalTemplate = utils.hasAttribute(templateBody, 'functional')
+  }
```

Comment:
This doesn't catch this case:

```vue
<template>…</template>

<script>
export default {
  functional: true,
  …
}
</script>
```

Please also add test cases for this.

Reply:
Hmm... I can't get it to work well using `functional: true` in SFC. Should it work?
(I've never used functional component before so I could be wrong.)

Reply:
Functional components without a template, but with a render function are initialized that way. See https://vuejs.org/v2/guide/render-function.html#Functional-Components

Reply:
Thank you for the explanation. I understand. I think the render function is already supported.

https://github.com/vuejs/eslint-plugin-vue/blob/ae34c6582981ffc31ec03176f4fc417ba2f2412d/tests/lib/rules/no-undef-properties.js#L198-L204

Reply:
Alright, then this fine for me.